src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -58,15 +66,54 @@ private static RequestAndRelatedRecords setRequestQueuePosition(
     }
   }
 
+  private static HttpResult<RequestAndRelatedRecords> refuseWhenRequestCannotBeFulfilled(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    RequestPolicy requestPolicy = requestAndRelatedRecords.getRequestPolicy();
+    RequestType requestType =  requestAndRelatedRecords.getRequest().getRequestType();
+
+    if(!requestPolicy.allowsType(requestType)) {
+      return failed(failure(
+        requestType.getValue() + " requests are not allowed for this patron and item combination", "requestType",
+        requestType.getValue()));
+    }
+    else {
+      return succeeded(requestAndRelatedRecords);
+    }
+  }
+
   private static HttpResult<RequestAndRelatedRecords> refuseWhenItemIsNotValid(
     RequestAndRelatedRecords requestAndRelatedRecords) {
 
     Request request = requestAndRelatedRecords.getRequest();
 
     if (!request.allowedForItem()) {
       return failed(failure(
-        String.format("Item is %s", request.getItem().getStatus()),
-        "itemId", request.getItemId()
+        String.format("%s requests are not allowed for %s item status combination", request.getRequestType().getValue() , request.getItem().getStatus().getValue()),
+        request.getRequestType().getValue(),
+        request.getItemId()
+      ));
+    }
+    else {
+      return succeeded(requestAndRelatedRecords);
+    }
+  }
+
+  private static HttpResult<RequestAndRelatedRecords> refuseWhenInvalidUserAndPatronGroup(
```

Comment:
This is a bit confusing to me. The `requesterId` value in the JSON is not required but we are effectively making it required in the code. Shouldn't we update the JSON schema to make `requesterId` required as well? Also, patron info is part of the user/requester. I suppose `patronGroup` could be not set when retrieved via mod-users, but there is no way for the request JSON to include it.

Also, shouldn't we use the actual JSON schema field names here? "User" is not a field that can be sent in the JSON. "requesterId" is. For "PatronGroupId", as I said above, there is no way to send this. In the users JSON schema this field is "patronGroup" so maybe we use that? But it would be better to indicate in the error message that the requester is not assigned a patron group/is misconfigured. "A valid patron group is required" seems to imply that this is something that can be corrected via the API.

Maybe we need to have the users JSON schema require a patron group so we don't have to worry about this here.

Also, do we actually do any patron group validation here? What happens is the patron group Id is set, but doesn't exist.

All this can probably be resolved later since I think in 99% of the cases this validation won't be an issue. Especially if we make `requesterId` required in the request schema.

Reply:
Good points, @mreno-EBSCO let's have an offline discussion about this question with @marcjohnson-kint and perhaps create a Jira for it